testing.rb

```diff
@@ -0,0 +1,313 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockNesting,Metrics/MethodLength,Metrics/ModuleLength,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity,Metrics/AbcSize
+
+# module Enumerable
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = self if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n])
+      n += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = self if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n], n)
+      n += 1
+    end
+    self
+  end
```

Comment:
Your `my_each_with_index` method presents the same problem as your `my_each` method. Just as before, please make sure your `my_each_with_index` is able to handle a hash with a proc in it's input.

Here's the error message returned by the automated tests:

```
 2) enumerables #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self
     Failure/Error: expect(my_each_with_index_output).to eq(each_with_index_output)
     
       expected: "Num: [:a, 1], idx: 0\nNum: [:b, 2], idx: 1\nNum: [:c, 3], idx: 2\nNum: [:d, 4], idx: 3\nNum: [:e, 5], idx: 4\n"
            got: "Num: , idx: 0\nNum: , idx: 1\nNum: , idx: 2\nNum: , idx: 3\nNum: , idx: 4\n"
     
       (compared using ==)
     
       Diff:
       @@ -1,6 +1,6 @@
       -Num: [:a, 1], idx: 0
       -Num: [:b, 2], idx: 1
       -Num: [:c, 3], idx: 2
       -Num: [:d, 4], idx: 3
       -Num: [:e, 5], idx: 4
       +Num: , idx: 0
       +Num: , idx: 1
       +Num: , idx: 2
       +Num: , idx: 3
       +Num: , idx: 4
```